### PR TITLE
Implement Send

### PR DESCRIPTION
**Pull request by dcposch**
ds

### CODE REVIEW

File: apps/daimo-mobile/src/logic/chain.ts
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/logic/chain.ts:161]: nit I think [`formatUnits`](https://viem.sh/docs/utilities/formatUnits.html) can also do this

File: apps/daimo-mobile/src/logic/chain.ts
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/logic/chain.ts:29]: nit and dont need to do here but I was thinking we can make @daimo/contracts generate and export some of the metadata of the erc20 contract separately like `address`/`decimals`/`name`/`symbol` etc. so we don't have to hardcode or fetch in-app. Will also make simpler to swap it out in future for us (and others who fork)

**Comment by dcposch** [comment]
### User op send issue

ds

all addresses (entry point, USDC, destination) look valid

File: apps/daimo-mobile/src/logic/chain.ts
Comment (dcposch) [reply apps/daimo-mobile/src/logic/chain.ts:141]: formatUnits doesn't have display decimals (so might output `"42.690418"` for 6-decimal USDC)

**Comment by dcposch** [comment]
### Fixed; send works now

TLDR;
- **To a send a user op, we need gas.** Either prepaid (ETH deposited in the EntryPoint on behalf of this specific account) or pay as you go (with a paymaster). 
- **We'll switch to paymaster once ready.** Kristof shipping a paymaster on Base soon. Package `@pimlico/erc20-paymaster` looks promising, still in active development.
- **See `scratchpad` package.** This is an easy place to try transactions, crypto, contract interactions without running a full Expo app.

File: apps/daimo-mobile/src/action/actStatus.ts
Comment (dcposch) [inline apps/daimo-mobile/src/action/actStatus.ts:20]: this is a nice reusable pattern

react hook which is meant to track a single user action, such as
- send
- add device 
- remove device
- create note
- redeem note

Those are all onchain ops. Pattern would fit future other ops (like waiting for approval from an onramp) as well.

- status `idle` > `loading` > `success`/`failure`
- at each point, a short updating message to show granular progress

i think we can use this hook for telemetry, to track anonymized / rolled-up performance and reliability data

File: apps/daimo-mobile/src/action/useCreateAccount.ts
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/action/useCreateAccount.ts:28]: I guess these logs/comments need to be updated (since no more deploy on this API?)

File: apps/scratchpad/src/index.ts
Comment (nalinbhardwaj) [inline apps/scratchpad/src/index.ts:98]: should this be `0.5`? don't we parse in @daimo/userop?

File: apps/daimo-mobile/src/view/screen/SendScreen.tsx
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/view/screen/SendScreen.tsx:144]: ah this was the intention for the `dryRun` parameter in @daimo/userop but I just realised that still asks for a signature: should change that behaviour in a seperate PR and use that

**Review by nalinbhardwaj — APPROVED** [review]
🔥

File: apps/scratchpad/src/index.ts
Comment (dcposch) [reply apps/scratchpad/src/index.ts]: fixed

File: apps/daimo-mobile/src/action/useCreateAccount.ts
Comment (dcposch) [reply apps/daimo-mobile/src/action/useCreateAccount.ts:28]: not yet but will soon.

this endpoint should do a single, atomic userOp that
- materializes the account
- infinite approves USDC to the paymaster
- claims the name

**Pull request merged**
